File: python/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dataset={}
        for index in range(len(nums)):
            if (target - nums[index]) in dataset:
                return [index, dataset[(target-nums[index])]]
            else:
                dataset[(nums[index])] = index

File: cpp/1.两数之和.cpp
#include "leetcode.h"

class Solution
{
public:
  vector<int> twoSum(vector<int>& nums, int target)
  {
    map<int, int> record;
    vector<int> res;
    for (auto i = 0; i < nums.size(); i++) {
      if (record.find(target - nums[i]) != record.end()) {
        return {i, record[target - nums[i]]};
      }
      record[nums[i]] = i;
    }
    return res;
  }
};
